File: algorithms/python/solution/S1TwoSum.py
# encoding:utf-8

# @author:jessenpan
# tag:array 


class TwoSum:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numWithIndexMap = {}
        for i, num in enumerate(nums):
            numWithIndexMap[num] = i
        result = []
        for j, oneOperatorNum in enumerate(nums):
            need_num = target - oneOperatorNum
            if (numWithIndexMap.__contains__(need_num)):
                otherOperatorIndex = numWithIndexMap[need_num]
            else:
                continue
            if (otherOperatorIndex != None and otherOperatorIndex != j):
                result.append(j)
                result.append(otherOperatorIndex)
                break
        return result

File: algorithms/java/src/main/java/org/jessenpan/leetcode/array/S1TwoSum.java
package org.jessenpan.leetcode.array;

import java.util.HashMap;
import java.util.Map;

/**
 * @author jessenpan
 * tag:array
 */
public class S1TwoSum {

    public int[] twoSum(int[] nums, int target) {
        int lengthOfNums = nums.length;
        Map<Integer, Integer> numWithIndexMap = new HashMap<Integer, Integer>(nums.length);
        for (int i = 0; i < lengthOfNums; i++) {
            numWithIndexMap.put(nums[i], i);
        }
        int[] resultArray = new int[2];
        for (int i = 0; i < lengthOfNums; i++) {
            int oneOperatorNum = nums[i];
            int needNum = target - oneOperatorNum;
            Integer otherOperatorIndex = numWithIndexMap.get(needNum);
            if (otherOperatorIndex != null && otherOperatorIndex != i) {
                resultArray[0] = i;
                resultArray[1] = otherOperatorIndex;
                break;
            }
        }
        return resultArray;
    }
}
